apps/web/src/app/(dashboard)/settings/billing/page.tsx

```diff
@@ -24,20 +27,37 @@ export default async function BillingSettingsPage() {
     redirect('/settings/profile');
   }
 
-  const [subscription, prices] = await Promise.all([
-    getSubscriptionByUserId({ userId: user.id }),
-    getPricesByInterval(),
+  if (!user.customerId) {
+    const stripeSession = await getStripeCustomerByUser(user);
+    user = stripeSession.user;
```

Comment:
Right now I understand the above code. I'd probably do the above

```suggestion
    user = await getStripeCustomerByUser(user).then(result => result.user);
```